router/core/context.go

```diff
@@ -261,6 +261,28 @@ type requestContext struct {
 	expressionContext expr.Context
 }
 
+func (c *requestContext) SetError(err error) {
+	c.error = err
+	c.expressionContext.Request.Error = err
+}
+
+func (c *requestContext) ResolveAnyExpressionWithWrappedError(expression *vm.Program) (any, error) {
+	// If an error exists already, wrap it and resolve the expression with the copied context
+	if c.expressionContext.Request.Error != nil {
+		return expr.ResolveAnyExpression(expression, expr.Context{
```

Comment:
This seems to be error prone when we add more fields to the context. How we catch an error here?

Reply:
Good catch, will take a look

Reply:
Done